Authentication/Authentication/Program.cs

```diff
@@ -0,0 +1,139 @@
+                    }
+                    catch (FileNotFoundException exception)
+                    {
+                        throw new UsersNotFoundException(exception.FileName);
+                    }
+                    finally
+                    {
+                        Console.WriteLine();
+                    }
+                }
+                else if (choice.Equals("Register", StringComparison.OrdinalIgnoreCase))
+                {
+                    Register(path);
+                    Console.WriteLine();
+                }
+                else
+                {
+                    Console.WriteLine("Invalid choice: {0}", choice);
+                    Console.WriteLine();
+                }
+            }
+
+        }
+
+        static string GetUserChoice()
+        {
+            Console.WriteLine("Would you like to:");
+            Console.WriteLine("Login");
+            Console.WriteLine("Register");
+            Console.WriteLine("Exit");
+            Console.Write("Your Choice: ");
+            var choice = Console.ReadLine();
+
+            return choice;
+        }
+
+        static string[] GetUserData()
+        {
+            var userData = new string[2];
+            Console.Write("Enter Your Username: ");
+            userData[0] = Console.ReadLine();
+
+            Console.Write("Enter Your Password: ");
+            userData[1] = Console.ReadLine();
+
+            return userData;
+        }
+
+        static void Login(string path)
+        {
+            var userData = GetUserData();
+            var userLoginString = $"{userData[0]} - {userData[1]}";
+
+            using var sr = new StreamReader(path);
+            string line;
+
+            while ((line = sr.ReadLine()) != null)
+            {
+                if (line.Equals(userLoginString))
+                {
+                    Console.WriteLine("Hello!");
+                }
+            }
+        }
+
+        static void Register(string path)
+        {
+            var userData = GetUserData();
+            var userRegisterString = $"{userData[0]} - {userData[1]}";
+
+            using (var sr = new StreamReader(path))
+            {
+                string line;
+                while ((line = sr.ReadLine()) != null)
+                {
+                    var userNameCheck = line.Split(' ');
+                    if (userNameCheck[0].Equals(userData[0]))
+                    {
+                        Console.WriteLine("User Name: {0} already exists", userData[0]);
+                        return;
+                    }
+                }
+            }
+
+            using var sw = new StreamWriter(path, true);
+            sw.Write(userRegisterString + Environment.NewLine);
+
+            // added these line so that the userRegisterString gets appended to the Users.txt present
+            // in the project folder as the during runtime it was only getting appended
+            // to the bin/dotnetcoreapp/debug/Users.txt file
```

Comment:
Why is that not ok?

Reply:
If I was adding users during the execution of the program it was only adding the register info in the Users.txt present in the debug folder (which is generated because of building the project) but when the project is ran again the unchanged Users.txt file present in project folder would replace the file present in the debug folder. So the new users who were registered during the execution of the program would not be recognised by Login( ) on the second run, so to combat this problem I added lines 119 and 120 which during the execution changed the Users.txt present inside the projects folder. So if you registered by running the programming and then exited the program and re-ran the program the users credentials used to registered can be used to Login( ) now.

Reply:
> when the project is ran again the unchanged Users.txt file present in project folder would replace the file present in the debug folder

You can select a property for "copy file" to "replace if newer". Your changed files will be newer, so they would not be replaced anymore.

Reply:
Oh okay, I didn't knew that, I'll make the changes.

Reply:
Changed it, Now it is working as intended.